.github/workflows/run_test_suite.yaml

```diff
@@ -1,35 +1,66 @@
-on: [push, pull_request]
+name: Tests and Checks
 
-name: Run test suite
+on:
+  push:
+    branches: [ main ]
+
+  pull_request:
+    branches: [ main ]
 
 jobs:
-  run-test-suite:
+  run-tests-checks:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        rust-toolchain:
+          - stable
+          - nightly
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        if: always()
-        with:
-          path: |
-            ~/.cargo
-            ./target
-          key: ${{ runner.os }}-cargo-artifacts-${{ hashFiles('**/Cargo.toml') }}
-      - name: 'Install environment packages'
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Cache Project
+        uses: Swatinem/rust-cache@v2
+
+      - name: Install Environment Packages
         run: |
           sudo apt-get update -qqy
           sudo apt-get install jq
-      - name: 'Run tests'
+
+      - name: Install Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          override: true
+          components: rustfmt, clippy
+          toolchain: ${{ matrix.rust-toolchain }}
+
+      - name: Check Format
+        uses: actions-rs/cargo@v1
```

Comment:
What is the advantage of using this action over running the command as a shell script?

Reply:
Tends to be more typical of rust projects/libs.